File: DP/1912.py
# https://www.acmicpc.net/problem/1912
# Name: 연속합
# Topic: Dynamic Programming


def solution(N):
    length = len(N)
    dp = {0: N[0]}
    ans = N[0]
    for i in range(1, length):
        if dp[i - 1] + N[i] > N[i]:
            dp[i] = dp[i - 1] + N[i]
        else:
            dp[i] = N[i]

        if ans < dp[i]:
            ans = dp[i]
    print(ans)

if __name__ == '__main__':
    n = input()
    a = input()
    a = [int(i) for i in a.split(' ')]
    solution(a)

File: DP/1912.cpp
// https://www.acmicpc.net/problem/1912
// Name: 연속 합
// Topic: Dynamic Programming

#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(0);

    int num;
    cin >> num;
    vector<int> v;
    for(int i = 0; i < num; i++) {
        int temp;
        cin >> temp;
        v.push_back(temp);
        if(i != 0)
            if(v[i - 1] > 0) v[i] += v[i - 1];
    }
    int max = *max_element(v.begin(), v.end());
    cout << max;
}